Mage.Sets/src/mage/cards/c/ConspiracyTheorist.java

```diff
@@ -0,0 +1,141 @@
+package mage.cards.c;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.costs.CompositeCost;
+import mage.abilities.costs.common.DiscardCardCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardsImpl;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.DiscardedCardsEvent;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.util.CardUtil;
+
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class ConspiracyTheorist extends CardImpl {
+
+    public ConspiracyTheorist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}");
+        
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SHAMAN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Whenever Conspiracy Theorist attacks, you may pay {1} and discard a card. If you do, draw a card.
+        this.addAbility(new AttacksTriggeredAbility(new DoIfCostPaid(new DrawCardSourceControllerEffect(1),
+            new CompositeCost(new ManaCostsImpl<>("{1}"), new DiscardCardCost(), "pay {1} and discard a card"))
+                .setText("you may pay {1} and discard a card. If you do, draw a card"), false));
+
+        // Whenever you discard one or more nonland cards, you may exile one of them from your graveyard. If you do, you may cast it this turn.
+        this.addAbility(new ConspiracyTheoristAbility());
+    }
+
+    private ConspiracyTheorist(final ConspiracyTheorist card) {
+        super(card);
+    }
+
+    @Override
+    public ConspiracyTheorist copy() {
+        return new ConspiracyTheorist(this);
+    }
+}
+
+class ConspiracyTheoristAbility extends TriggeredAbilityImpl {
+
+    ConspiracyTheoristAbility() {
+        super(Zone.BATTLEFIELD, null);
+    }
+
+    ConspiracyTheoristAbility(ConspiracyTheoristAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DISCARDED_CARDS;
```

Comment:
Must be checked -- is it catches only multiple discarded cards or can catch a single card (I dont remember is it fixed or not -- maybe some effects calls single discard, so your trigger will fail but it must catch it too by rules).

Reply:
This one should be fine, it fires for a single discard